DM_2/users/users_model.js

```diff
@@ -0,0 +1,41 @@
+const DefaultError = require('../exceptions/defaultError');
+const Users = require('../sequelize_models/Users');
+
+module.exports = class UsersModel {
+  constructor(req, res) {
+    this.req = req;
+    this.res = res;
+  }
+
+  async checkIsUser() {
+    try {
+      const resultAuth = await Users.findAll({
+        attributes: ['id'],
+        where: {
+          phone: this.req.headers.userphone,
+          password: this.req.headers.userpassword,
+        },
+      });
+      return resultAuth.length !== 0;
+    } catch (err) {
+      throw new DefaultError(err);
+    }
+  }
+
+  async regUser() {
+    try {
+      await this.checkIsUser();
```

Comment:
if user exist - you should raise an error and stop function

Reply:
agreed. Done